cinema/tests/test_movie_viewset.py

```diff
@@ -0,0 +1,161 @@
+from django.urls import reverse
+from rest_framework.exceptions import ErrorDetail
+from rest_framework.test import APITestCase, APIClient
+from rest_framework import status
+from cinema.models import Movie, Genre, Actor
+from django.contrib.auth import get_user_model
+from rest_framework.authtoken.models import Token
+from io import BytesIO
+from PIL import Image
+
+User = get_user_model()
+
+
+def create_genre(name: str) -> Genre:
+    return Genre.objects.create(name=name)
+
+
+def create_actor(first_name: str, last_name: str) -> Actor:
+    return Actor.objects.create(first_name=first_name, last_name=last_name)
+
+
+def create_movie(title: str, genres=None, actors=None) -> Movie:
+    movie = Movie.objects.create(
+        title=title, description="Sample description", duration=120
+    )
+    if genres:
+        movie.genres.set(genres)
+    if actors:
+        movie.actors.set(actors)
+    return movie
+
+
+def generate_image_file():
+    image = Image.new("RGB", (100, 100))
+    byte_arr = BytesIO()
+    image.save(byte_arr, format="PNG")
+    byte_arr.seek(0)
+    return byte_arr
+
+
+class MovieViewSetTests(APITestCase):
```

Comment:
It's a good practice to separate the test data creation and the actual test cases into different classes. You can create a base class to setup common data for all test cases, then each test case class can inherit from this base class. This makes the test cases cleaner and easier to manage.